packages/apollo-gateway/src/cache.ts

```diff
@@ -0,0 +1,64 @@
+import { URL, format } from 'url';
+import { CacheManager } from 'make-fetch-happen';
+import { Request, Response } from 'apollo-server-env';
+import { InMemoryLRUCache } from 'apollo-server-caching';
+
+const MAX_SIZE = 5 * 1024 * 1024; // 5MB
+
+/**
+ * @see: https://github.com/npm/make-fetch-happen/blob/master/cache.js
+ */
+function cacheKey(request: Request) {
+  const parsed = new URL(request.url);
+  const key = `gateway:request-cache:${format({
+    protocol: parsed.protocol,
+    slashes: true,
+    port: parsed.port,
+    hostname: parsed.hostname,
+    pathname: parsed.pathname,
+  })}`;
+  return key;
+}
```

Comment:
A simpler approach was sufficient, and I squeezed the `request.method` in there as well like we discussed since it really couldn't hurt anything.
40acac8